src/stores/connectors/eth1/restore.go

```diff
@@ -19,26 +19,29 @@ func (c Connector) Restore(ctx context.Context, addr ethcommon.Address) error {
 		return err
 	}
 
-	acc, err := c.db.GetDeleted(ctx, addr.Hex())
-	if err != nil {
-		return err
+	_, err = c.Get(ctx, addr)
+	if err == nil {
+		return nil
 	}
 
-	err = c.db.RunInTransaction(ctx, func(dbtx database.ETH1Accounts) error {
-		err = dbtx.Restore(ctx, addr.Hex())
+	acc, err := c.db.GetDeleted(ctx, addr.Hex())
```

Comment:
If key exists already, should so return as you do in line 22-24, but rest of code should have remained the same way as it was